File: Data Structures & Algorithms/two-integer-sum/submission-20.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> seen{};
        seen[nums[0]] = 0;

        for (auto i = 1; i < nums.size(); ++i) {
            int addend = target - nums[i];
            if (seen.find(addend) != seen.end()) {
                return { seen[addend], i };
            }
            seen[nums[i]] = i;
        }

        return {};
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-20.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        seen = set(nums)
        
        longestLen = 0
        for num in nums:
            if num - 1 in seen:
                continue

            length = 1
            while num + length in seen:
                length += 1

            longestLen = max(longestLen, length)

        return longestLen
        